src/cmake-tools.ts

```diff
@@ -406,16 +544,39 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       }
       switch (communicationMode) {
         case fileApi:
-          drv = await CMakeFileApiDriver
-              .create(cmake, this.workspaceContext.config, kit, workspace, preConditionHandler, preferredGenerators);
+          drv = await CMakeFileApiDriver.create(cmake, this.workspaceContext.config,
+                                                this.useCMakePresets,
+                                                this.activeKit,
+                                                this.configurePreset,
+                                                this.buildPreset,
+                                                this.testPreset,
+                                                workspace,
+                                                preConditionHandler,
+                                                preferredGenerators);
           break;
         case serverApi:
-          drv = await CMakeServerClientDriver
-              .create(cmake, this.workspaceContext.config, kit, workspace, preConditionHandler, preferredGenerators);
+          drv = await CMakeServerClientDriver.create(cmake,
+                                                     this.workspaceContext.config,
+                                                     this.useCMakePresets,
+                                                     this.activeKit,
+                                                     this.configurePreset,
+                                                     this.buildPreset,
+                                                     this.testPreset,
+                                                     workspace,
+                                                     preConditionHandler,
+                                                     preferredGenerators);
           break;
         default:
-          drv = await LegacyCMakeDriver
-            .create(cmake, this.workspaceContext.config, kit, workspace, preConditionHandler, preferredGenerators);
+          drv = await LegacyCMakeDriver.create(cmake,
+                                               this.workspaceContext.config,
+                                               this.useCMakePresets,
+                                               this.activeKit,
+                                               this.configurePreset,
+                                               this.buildPreset,
+                                               this.testPreset,
+                                               workspace,
+                                               preConditionHandler,
+                                               preferredGenerators);
         }
     } finally { this._statusMessage.set(localize('ready.status', 'Ready')); }
 
```

Comment:
I can't add a comment on line 588, but there is a problem with `registerTaskProvider` when using presets because `this.tasksBuildCommandDrv` will try to generate the build command, but the build preset is not yet set, so it fails and returns null.

Reply:
I didn't notice this function. I'll do reregisterTaskProvider in setBuildPresets.